packages/flame_forge2d/lib/contact_callbacks.dart

```diff
@@ -1,93 +1,55 @@
-import 'package:forge2d/forge2d.dart';
-
-import 'body_component.dart';
-
-class ContactTypes<T1, T2> {
-  // If o1 is, or inherits from, T1 or T2
-  bool has(Object o1) => o1 is T1 || o1 is T2;
-  bool hasOne(Object o1, Object o2) => has(o1) || has(o2);
-
-  // Only makes sense to call with objects that you know is in [T1, T2]
-  bool inOrder(Object o1, Object o2) => o1 is T1 && o2 is T2;
-
-  // Remember that this is not symmetric, it checks if the types in `o1` and
-  // `o2` are the same or inherits from the types in `other`
-  bool match(Object o1, Object o2) =>
-      (o1 is T1 && o2 is T2) || (o2 is T1 && o1 is T2);
-}
-
-typedef ContactCallbackFun = void Function(Object a, Object b, Contact contact);
-
-abstract class ContactCallback<Type1, Type2> {
-  ContactTypes<Type1, Type2> types = ContactTypes<Type1, Type2>();
-
-  void begin(Type1 a, Type2 b, Contact contact) {}
-  void end(Type1 a, Type2 b, Contact contact) {}
-  void preSolve(Type1 a, Type2 b, Contact contact, Manifold oldManifold) {}
-  void postSolve(Type1 a, Type2 b, Contact contact, ContactImpulse impulse) {}
-}
-
-class ContactCallbacks extends ContactListener {
-  final List<ContactCallback> _callbacks = [];
-
-  void register(ContactCallback callback) {
-    _callbacks.add(callback);
+import 'flame_forge2d.dart';
+
+/// Used to listens to a [BodyComponent] contacts events.
+///
+/// Contacts events occur whenever two [Fixture] meet each other.
+///
+/// To react to contacts you should assign [ContactCallbacks] to a [Body]'s
+/// userData or/and to [Fixture]'s userData.
+/// {@macro flame_forge2d.world_contact_listener.algorithm}
+class ContactCallbacks {
+  /// Called when two [Fixture]s begin to overlap.
+  ///
+  /// It is called for sensors and non-sensors.
+  void beginContact(Object other, Contact contact) {
+    onBeginContact?.call(other, contact);
   }
 
-  void deregister(ContactCallback callback) {
-    _callbacks.remove(callback);
+  /// Called when two [Fixture]s cease to overlap.
+  ///
+  /// It is called for sensors and non-sensors.
+  ///
+  /// It may be called when a body is destroyed.
```

Comment:
May?

Reply:
I got this information from the docs, can't locate exactly where.

```dart
final wasTouching = (flags & touchingFlag) == touchingFlag;
...
if (wasTouching && !touching) {
  listener.endContact(this);
}
```

Need to further investigate this logic since I'm not sure if we are updating flags in Forge2D when we destroy a body from the world. Also, there is a chance this could depend if they cease contact and are destroyed in the same step cycle.

Regardless, for the time being, I think is better if we remove this line of documentation and delegate this to Forge2D documentation once provided.

```suggestion
```

Reply:
I'm fairly sure that the flags are updated otherwise other things would break too.
It doesn't matter that `destroyBody` does `assert(!isLocked)`?

I think that you can just remove the line about the destruction since it is implied that the contact then ends and then all the lines can be merged to something like;
> Called when two [Fixture]s, cease to be in contact.

I don't think that we have to mention sensors specifically, just like we don't mention static bodies specifically.

Reply:
Removed 👍 

Personally, I find the sensor information useful and is straight from Box2D. I would like to keep them but if you think we definitely shouldn't let me know and I'll get rid of them.